server/src/main/java/org/elasticsearch/index/mapper/VersionFieldMapper.java

```diff
@@ -76,4 +76,6 @@ public void postParse(ParseContext context) {
     protected String contentType() {
         return CONTENT_TYPE;
     }
+
+    public static final VersionFieldMapper INSTANCE = new VersionFieldMapper();
```

Comment:
we use these static instances only for the tombstone handling? Is there no chance to use the same instances that MapperService holds? I guess hard to make them static if we do so?

Reply:
I've changed their TypeParsers to return the static instances as well - they're entirely stateless, we only need a single copy of each one.

Reply:
looks good, thanks